pkg/asset/machines/azure/machines.go

```diff
@@ -105,6 +107,22 @@ func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string
 		managedIdentity = ""
 	}
 
+	managedDisk := machineapi.ManagedDiskParameters{
+		StorageAccountType: mpool.OSDisk.DiskType,
+		EncryptionType:     string(compute.EncryptionAtRestWithPlatformKey),
+	}
+
+	var securityProfile *machineapi.SecurityProfile
+	if mpool.EncryptionAtHost {
+		securityProfile = &machineapi.SecurityProfile{
+			EncryptionAtHost: &mpool.EncryptionAtHost,
+		}
+	}
+
+	if mpool.OSDisk.EncryptionType != "" {
+		managedDisk.EncryptionType = mpool.OSDisk.EncryptionType
```

Comment:
There is no such thing as [`encryptionType` in Azure API](https://docs.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update#manageddiskparameters) in the VM create or update endpoint.

We need `diskEncryptionSet` as it was in #5134. And I think we don't need https://github.com/openshift/api/pull/1118.

Or am I missing something?

Reply:
I will be reworking this PR. Thanks for all your feedback.